### DISCUSSION TITLE
In a network game, Bots clean up a stack of items marked as untouchable.

### DISCUSSION THREAD

**Discussion by pavel4win** [FakeFishGames/Barotrauma#14427] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

If you mark a stack of items as untouchable by the bots.
And then throw this stack on the floor.
And then give to any bot the order to clean up the boat.
Bot will clean up the stack of items in the closet.
If you mark a stack of items as untouchable by the bots.
And then you split the stack and throw the items on the floor one by one.
Then the bots will not clean up the items.
Is this a bug or a feature?

### Reproduction steps

I tested this with many different items. For example, stack of saline, iron, fruit.

1. Start multiplayer game with bots.
2. mark any stack of items as untouchable by the bots.
3. Then throw this stack on the floor.
4. And give to any bot the order to clean up the boat.
5. Bot will clean up the stack of items in the closet.
But:
If you mark a stack of items as untouchable by the bots.
And then you split the stack and throw the items on the floor one by one.
Then the bots will not clean up the items.
It is important that the items do not fall in one place, you need to take a step after each drop. Otherwise, it will clean up all or almost all the items.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted using a dedicated server

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! This did turn out to be a bug, similar to the past issues of stacks not getting marked for deconstruction. Fixed in https://github.com/FakeFishGames/Barotrauma-development/pull/5614/commits/5f1c21051216acee042ea5d09bec740da1c3d1fb (commit in our private repo), fix coming up in a future build!

**Comment by Rokvach** [comment]
Tested the fix, working correctly. Fix coming in the next update. Closing discussion.
